pkgs/development/python-modules/torch/default.nix

```diff
@@ -344,17 +344,15 @@ in buildPythonPackage rec {
 
     # the following are required for tensorboard support
     pillow six future tensorboard protobuf
+
+    # ROCm build and `torch.compile` requires openai-triton
+    openai-triton
+
+    # torch/csrc requires `pybind11` at runtime
+    pybind11
```

Comment:
To put it another way, is it ever required when using the python modules in `torch.*`? If it's really only required for the c++ stuff, maybe we should put this into `nix-support/propagated-build-inputs` in a separate output. That said, I think at the moment our `torch` isn't very smart about what to put in which output (the c++ dependencies leak into python environments, and python dependencies leak into the c++ dev environments)

Reply:
It is likely required for ``torch.compile``.

Reply:
Specifically, ``fairseq`` needs it.
This likely extends to user applications.

Reply:
What do I need to `import` and call to reproduce the error?

Reply:
Actually now I'm not sure.
It might be only for C++ stuff, in which case we can just move pybind11 to -dev's propagated build inputs.
/nix/store/*-python3.11-torch-2.0.1-dev/include/torch/csrc/Exceptions.h

Reply:
After looking into this, ``nix-support`` is only in ``-dev`` anyway, so doing any of that would just be an effective no-op with a massive rebuild for no reason.
Unless you have a better idea on what to do, I think this matter is resolved.

Reply:
> After looking into this, nix-support is only in -dev anyway

Yes, that's what I meant, and I think I want to start writing `nix-support/propagated-build-inputs` manually so that we can put the C++ and python stuff into separate outputs

Reply:
I think that would be good.
I'm hesitant to do that however in this PR, since the goal of this PR is to fix everything.
I'll look into doing that in a separate PR, since it will likely require me changing more than a few packages to account for the change.